src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyExecutionManager.cs

```diff
@@ -96,7 +98,10 @@ public virtual int StartTestRun(TestRunCriteria testRunCriteria, ITestRunEventsH
                     testSources = testRunCriteria.Tests.GroupBy(tc => tc.Source).Select(g => g.Key);
                 }
 
-                this.isCommunicationEstablished = this.SetupChannel(testSources, this.cancellationTokenSource.Token);
+                lock (locObject)
```

Comment:
why do we need a lock while setting up channel? Is it possible that we try to setup communication even though test run times out?